prove_withdrawal.go

```diff
@@ -166,153 +160,141 @@ func (epp *EigenPodProofs) ProveWithdrawals(
-	if err != nil {
-		return nil, err
+	var withdrawalExecutionPayloadFieldRoots []phase0.Root
+	var withdrawals []*capella.Withdrawal
+	var withdrawalFields []Bytes32
+	var withdrawalIndex uint64
+	var timestamp uint64
+	if withdrawalBlock.Version == spec.DataVersionDeneb {
+		start = time.Now()
+		// prove the execution payload against the withdrawal block header
+		withdrawalProof.ExecutionPayloadProof, withdrawalProof.ExecutionPayloadRoot, err = beacon.ProveExecutionPayloadAgainstBlockHeaderDeneb(withdrawalBlockHeader, withdrawalBlock.Deneb.Body)
+		if err != nil {
+			return nil, nil, err
+		}
+		log.Info().Msgf("time to prove execution payload against block header: %s", time.Since(start))
+
+		// calculate execution payload field roots
+		withdrawalExecutionPayloadFieldRoots, err = beacon.ComputeExecutionPayloadFieldRootsDeneb(withdrawalBlock.Deneb.Body.ExecutionPayload)
+		if err != nil {
+			return nil, nil, err
+		}
+		withdrawals = withdrawalBlock.Deneb.Body.ExecutionPayload.Withdrawals
+		withdrawalIndex = GetWithdrawalIndex(validatorIndex, withdrawals)
+		withdrawalFields = ConvertWithdrawalToWithdrawalFields(withdrawalBlock.Deneb.Body.ExecutionPayload.Withdrawals[withdrawalIndex])
+		timestamp = withdrawalBlock.Deneb.Body.ExecutionPayload.Timestamp
+	} else if withdrawalBlock.Version == spec.DataVersionCapella {
+		start = time.Now()
+		// prove the execution payload against the withdrawal block header
+		withdrawalProof.ExecutionPayloadProof, withdrawalProof.ExecutionPayloadRoot, err = beacon.ProveExecutionPayloadAgainstBlockHeaderCapella(withdrawalBlockHeader, withdrawalBlock.Capella.Body)
+		if err != nil {
+			return nil, nil, err
+		}
+		log.Info().Msgf("time to prove execution payload against block header: %s", time.Since(start))
+
+		// calculate execution payload field roots
+		withdrawalExecutionPayloadFieldRoots, err = beacon.ComputeExecutionPayloadFieldRootsCapella(withdrawalBlock.Capella.Body.ExecutionPayload)
+		if err != nil {
+			return nil, nil, err
+		}
+		withdrawals = withdrawalBlock.Capella.Body.ExecutionPayload.Withdrawals
+		withdrawalIndex = GetWithdrawalIndex(validatorIndex, withdrawals)
+		withdrawalFields = ConvertWithdrawalToWithdrawalFields(withdrawalBlock.Capella.Body.ExecutionPayload.Withdrawals[withdrawalIndex])
+
+		timestamp = withdrawalBlock.Capella.Body.ExecutionPayload.Timestamp
+	} else {
+		return nil, nil, errors.New("unsupported version")
 	}
-	log.Info().Msgf("time to prove execution payload against block header: %s", time.Since(start))
 
-	start = time.Now()
-	// calculate execution payload field roots
-	withdrawalExecutionPayloadFieldRoots, err := beacon.ComputeExecutionPayloadFieldRootsCapella(withdrawalBlock.Body.ExecutionPayload)
+	oracleBeaconStateHistoricalSummaries, err := beacon.HistoricalSummaries(oracleBeaconState)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	err = epp.proveWithdrawal(
 		withdrawalProof,
```

Comment:
this arg list keeps growing which makes code hard to maintain; in general, if function needs more than 3 args, it is recommend to wrap all args into a container struct with required args

Reply:
thsi shouldn't grow beyonf three